src/jedi_bundle/utils/git.py

```diff
@@ -122,34 +129,59 @@ def get_url_and_branch(logger, github_orgs, repo_url_name, default_branch, user_
             # Track first instance of finding the default branch. But do not exit when it's first
             # found so that other organizations can be checked for the user branch.
             if not found_default_branch:
-                if repo_has_branch(logger, github_url, default_branch, is_tag_in):
+                if repo_has_branch(logger, github_url, default_branch, is_tag_in, is_commit_in):
                     found_default_branch = True
                     repo_url_found = True
                     repo_url_to_use = github_url
                     repo_branch_to_use = default_branch
                     is_tag = is_tag_in
+                    is_commit = is_commit_in
 
-    return repo_url_found, repo_url_to_use, repo_branch_to_use, is_tag
+    return repo_url_found, repo_url_to_use, repo_branch_to_use, is_tag, is_commit
 
 
 # --------------------------------------------------------------------------------------------------
 
 
-def clone_git_repo(logger, url, branch, target, is_tag):
+def clone_git_repo(logger, url, branch, target, is_tag, is_commit):
 
     # Check if directory already exists
     if not os.path.exists(target):
 
-        # Command to check if branch exists and pass exit code back
-        git_clone_cmd = ['git', 'clone', '--recursive', '-b', branch, url, target]
-
-        # Run command
-        subprocess_run(logger, git_clone_cmd, True)
+        if is_commit:
+            # Clone repo
+            git_clone_cmd = ['git', 'clone', url, target]
+            subprocess_run(logger, git_clone_cmd, True)
+            parent_dir = os.path.abspath(os.curdir)
+            os.chdir(target)
+            # Run check for commit
+            try:
+                git_check_cmd = ['git', 'cat-file', '-t', branch]
+                commit_valid = subprocess.check_output(git_check_cmd)
+                if 'commit' in str(commit_valid, 'utf-8'):
+                    git_clone_cmd = ['git', 'checkout', branch]
+                    subprocess_run(logger, git_clone_cmd, True)
+                    os.chdir(parent_dir)
+            except Exception:
+                os.chdir('../')
```

Comment:
Probably better to avoid `os.chdir(...)`. One option is to force git to assume a particular working directory with `-C pwd` (see `man git`):

```suggestion
            parent_dir = os.path.abspath(os.curdir)
            # Run check for commit
            try:
                git_check_cmd = ['git', '-C', target, 'cat-file', '-t', branch]
                commit_valid = subprocess.check_output(git_check_cmd)
                if 'commit' in str(commit_valid, 'utf-8'):
                    git_clone_cmd = ['git', '-C', target, 'checkout', branch]
                    subprocess_run(logger, git_clone_cmd, True)
            except Exception:
```

Alternatively, `subprocess.run` accepts `cwd` as an argument; e.g., `subprocess.run("pwd", cwd = "/path/to/other/place")` will print ("/path/to/other/place"). That works with `subprocess.run` --- not sure if kwargs are propagated through to `subprocess_run`.